flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/hybrid/HybridSourceSplit.java

```diff
@@ -19,33 +19,45 @@
 package org.apache.flink.connector.base.source.hybrid;
 
 import org.apache.flink.api.connector.source.SourceSplit;
+import org.apache.flink.core.io.SimpleVersionedSerializer;
 
+import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 import java.util.Objects;
 
 /** Source split that wraps the actual split type. */
 public class HybridSourceSplit implements SourceSplit {
 
-    private final SourceSplit wrappedSplit;
+    private final byte[] wrappedSplitBytes;
+    private final int wrappedSplitSerializerVersion;
```

Comment:
```suggestion
    private final int serializerVersion;
```

Reply:
Considered that as well and prefer the verbose name to make clear that this is the serializer version for the underlying state vs that of `HybridSourceSplit`.